golang/tracing.go

```diff
@@ -0,0 +1,76 @@
+package tchannel
+
+import (
+	"fmt"
+	"math/rand"
+
+	"golang.org/x/net/context"
+)
+
+// Span represents Zipkin-style span
+type Span struct {
+	traceID  uint64
+	parentID uint64
+	spanID   uint64
+	flags    byte
+}
+
+func (s Span) String() string {
+	return fmt.Sprintf("TraceID=%d,ParentID=%d,SpanID=%d", s.traceID, s.parentID, s.spanID)
+}
+
+const (
+	tracingFlagEnabled byte = 0x01
+	tracingKey              = "tracing"
+)
+
+// NewRootSpan creates a new top-level Span for a call-graph within the provided context
+func NewRootSpan() *Span {
+	return &Span{traceID: uint64(rand.Int63())}
```

Comment:
Do we need to seed `rand` before we use it? I think otherwise it generates the same numbers every time:

```go
~ ❯ cat test.go
package main

import (
	"fmt"
	"math/rand"
)

func main() {
	fmt.Println(rand.Int63())
}
~ ❯ go build test.go
~ ❯ ./test
5577006791947779410
~ ❯ ./test
5577006791947779410
~ ❯ ./test
5577006791947779410
~ ❯ go run test.go
5577006791947779410
```